test/integration/scheduler_perf/scheduler_test.go

```diff
@@ -253,3 +232,123 @@ func schedulePods(config *testConfig) int32 {
 		time.Sleep(1 * time.Second)
 	}
 }
+
+// mutateNode returns the strategy needed for creation of nodes.
+// TODO: It should take the nodespec and return the modified version of it.
+func (nodeAffinity *nodeAffinity) mutateNode(numNodes int) []testutils.CountToStrategy {
+	numGroups := nodeAffinity.numGroups
+	nodeAffinityKey := nodeAffinity.nodeAffinityKey
+	nodeStrategies := make([]testutils.CountToStrategy, 0, numGroups)
+	for i := 0; i < numGroups; i++ {
+		nodeStrategies = append(nodeStrategies, testutils.CountToStrategy{
+			Count:    numNodes / numGroups,
+			Strategy: testutils.NewLabelNodePrepareStrategy(nodeAffinityKey, fmt.Sprintf("%v", i)),
+		})
+	}
+	return nodeStrategies
+}
+
+// mutatePod returns the list of pods after mutating the pod spec based on predicates and priorities.
+func (nodeAffinity *nodeAffinity) mutatePod(numPods int, pod *v1.Pod) []*v1.Pod {
+	numGroups := nodeAffinity.numGroups
+	nodeAffinityKey := nodeAffinity.nodeAffinityKey
+	podList := make([]*v1.Pod, 0, numGroups)
+	for i := 0; i < numGroups; i++ {
+		pod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "sched-perf-node-affinity-pod-",
+			},
+			Spec: testutils.MakePodSpec(),
+		}
+		pod.Spec.Affinity = &v1.Affinity{
+			NodeAffinity: &v1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+					NodeSelectorTerms: []v1.NodeSelectorTerm{
+						{
+							MatchExpressions: []v1.NodeSelectorRequirement{
+								{
+									Key:      nodeAffinityKey,
+									Operator: nodeAffinity.Operator,
+									Values:   []string{fmt.Sprintf("%v", i)},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		podList = append(podList, pod)
+	}
+	return podList
+}
+
+// readInConfiguration reads the input parameters for every predicate and priority that are needed for test run.
+// TODO: As of now, returning configs hardcoded, need to read from yaml or some other file.
```

Comment:
To be clear ; I originally figured we should have default behavior for readInConfig ... but am now thinking it really should be moved out to be done later ;  good point that the MVP shouldn't hint at a user interface .